monkey/tests/unit_tests/infection_monkey/credential_collectors/test_mimikatz_collector.py

```diff
@@ -110,3 +117,23 @@ def test_pypykatz_result_parsing_no_secrets(monkeypatch):
     collected_credentials = collect_credentials()
     assert len(collected_credentials) == 1
     assert collected_credentials == expected_credentials
+
+
+def test_mimikatz_credentials_stolen_event_published(monkeypatch):
+    mock_event_queue = MagicMock(spec=IEventQueue)
+    mock_pypykatz_handler = Mock()
+    monkeypatch.setattr(
+        "infection_monkey.credential_collectors.mimikatz_collector.pypykatz_handler",
+        mock_pypykatz_handler,
+    )
+
+    mimikatz_credential_collector = MimikatzCredentialCollector(mock_event_queue)
+    collected_credentials = mimikatz_credential_collector.collect_credentials()
+
+    mock_event_queue.publish.assert_called_once()
+
+    mock_event_queue_call_args = mock_event_queue.publish.call_args[0][0]
+
+    assert isinstance(mock_event_queue_call_args, CredentialsStolenEvent)
+    assert mock_event_queue_call_args.tags == MIMIKATZ_EVENT_TAGS
+    assert mock_event_queue_call_args.stolen_credentials == collected_credentials
```

Comment:
stolen_credentials and collected_credentials are both empty because the `mock_pypykatz_handler` returns no data. There's already a `patch_pypykatz()` function we can use to achieve this.